### DISCUSSION TITLE
Hey Awesome Devs who Created this Plugin. I finally found a Python Linux-amd64 IDE that is scaleable for my Requirements. Question regarding Creating Additional "File Types" when creating a New Python File.

### DISCUSSION THREAD

**Discussion by KastningBrandon** [albilu/netbeansPython#107] [General]
Hey Awesome Devs who Created this Plugin. I finally found a Python Linux-amd64 IDE that is scaleable for my Requirements. Question regarding Creating Additional "File Types" when creating a New Python File.

I want to make several Template Files that will be in the Menu of "File -> "New File" -> "Categories" -> "Python" -> "File Types".

Currently there are only the following:

1) "Empty Python File"
2) "Python File"
3) "Python PyTest File"
4) "Python Unit Test File"

Where are these templates stored? Are they hard-coded into the plugin that is compiled into your Release File "python-RELEASE200-1.9.nbm" (I'm unfamiliar with that file extension).

Or is this something in my debian-12.5.0-amd64 Linux Operating System that I can modify on the file system on flat configuration files? 

I would appreciate a good key in the right direction to achieve what I am needing/requiring. If my requirements/needs aids your project; I do not know C/C++ at this time in my existence here. I'm just starting out with Computer Science 101; bash scripting, Incredibly Beginner Python3x. However I have proven to have potential once I get my IDE worked out the way I need it and learn the fundamentals of each function/requirement of my Programming Needs.

Thank you for your time and your project! It's very awesome! Apache NetBeans 20 is better than anything i've used so far and with your plugin; very hopeful for my pathway.

Best Regards,

Brandon Kastning

**Comment by KastningBrandon** [comment]
albilu,

Where the .Py Files are located... Now to locate the Dialog Window Menu ".xml's or .json's" that contains the list of the .py files? & Then how to compile and build an NPM for a re-install of the Plugin which I could publish on my Github. (Which if I can't figure it out; What do you require for me to include in my re-compile; I want to add lots of "Data Science/Bulk Data Parsing" Python Templates that could benefit many. 

netbeansPython/src/main/resources/org/netbeans/modules/python/

- EmptyPython.py
- Python.py
- PythonPytest.py
- PythonUnittest.py

Best Regards,

Brandon Kastning
Arlington, WA
Mystery Babylon According to Revelation (KJV/KJV1611) 
(Overtly Overthrown United States of America, The Constitutional Republic)

**Comment by KastningBrandon** [comment]
albilu,

Update 04/14/2024 - 11:08PM

Utilizing Eclipse-IDE to search through the Source Code... the Dialog Window for "New File" and the Bottom "Description" Box that changes Text based on the "File Types" (4 Default Choices as Mapped/Stated Above) source file is ...

/netbeansPython/src/main/resources/org/netbeans/modules/python/PythonDescription.html


With this discovery... I need to find the file that contains the mapping between the .py & the .html's ... My guess at this point is in a .xml file.

Brandon Kastning
Arlington, WA
Mystery Babylon According to Revelation (KJV/KJV1611)
(Overtly Overthrown United States of America, The Constitutional Republic)

**Comment by albilu** [comment]
You can achieve this via: Tools -> Templates -> Python => Duplicate One of the default template and modify as you want.

Then it will be available in the New File creation menu
